robots/commenter/main.go

```diff
@@ -209,13 +216,31 @@ func makeCommenter(comment string, useTemplate bool) func(meta) (string, error)
 	}
 }
 
-func run(c client, query, sort string, asc, random bool, commenter func(meta) (string, error), ceiling int) error {
-	log.Printf("Searching: %s", query)
-	issues, err := c.FindIssues(query, sort, asc)
-	if err != nil {
-		return fmt.Errorf("search failed: %v", err)
+func run(c client, queries, sort string, asc, random bool, commenter func(meta) (string, error), ceiling int, labelsUpdated time.Duration) error {
+	var issues []github.Issue
```

Comment:
I would have used a set (map with bool value) to track which ones we already have. So we save a lot of for loops